terminator/lib/transaction-monitor-workflow.js

```diff
@@ -0,0 +1,62 @@
+/** @import {WorkflowEvent, WorkflowStep} from 'cloudflare:workers' */
+import { WorkflowEntrypoint } from 'cloudflare:workers'
+import { getChainClient as defaultGetChainClient } from './chain.js'
+
+/**
+ * Workflow that monitors a transaction and starts cancel workflow if it takes
+ * too long
+ */
+export class TransactionMonitorWorkflow extends WorkflowEntrypoint {
+  /**
+   * @param {WorkflowEvent} event
+   * @param {WorkflowStep} step
+   */
+  async run(
+    { payload: { transactionHash } },
+    step,
+    { getChainClient = defaultGetChainClient } = {},
+  ) {
+    try {
+      // Wait for transaction receipt with timeout
+      await step.do(
+        `wait for transaction receipt ${transactionHash}`,
+        {
+          timeout: `5 minutes`,
+          retries: {
+            limit: 3,
+          },
+        },
+        async () => {
+          const { publicClient } = getChainClient(this.env)
+          return await publicClient.waitForTransactionReceipt({
+            hash: transactionHash,
+          })
+        },
+      )
+
+      await step.do(
+        'clean terminate_service_tx_hash when transaction is confirmed',
+        {},
+        async () => {
+          this.env.DB.prepare(
+            'UPDATE data_sets SET terminate_service_tx_hash = NULL WHERE terminate_service_tx_hash = ?',
```

Comment:
```suggestion
            'UPDATE data_sets SET terminate_service_tx_hash = NULL, with_cdn = 0 WHERE terminate_service_tx_hash = ?',
```

should we do this here as well to guarantee consistency?